subt/breadcrumbs.py

```diff
@@ -28,22 +29,33 @@ def should_deploy(self, xyz):
                 return False
         return True
 
+    def deploy(self):
+        if self.num_avail > 0:
+            self.publish('deploy', [])  # ROS std_msg/Empty expects empty list as input
+            self.num_avail -= 1
+            return True
+        return False
+
     def on_sim_time_sec(self, data):
         if self.start_time is None:
             self.start_time = data
             if self.step_time is not None:
                 self.next_deploy_time = self.start_time + self.step_time
         if self.next_deploy_time is not None and data > self.next_deploy_time:
             self.next_deploy_time += self.step_time
-            self.publish('deploy', [])  # ROS std_msg/Empty expects empty list as input
+            self.deploy()
 
     def on_pose3d(self, data):
         xyz, quat = data
-        if self.should_deploy(xyz):
+        if self.should_deploy(xyz) and self.deploy():
             self.locations.append(xyz)
-            self.publish('deploy', [])
             self.publish('location', xyz)
 
+    def on_external(self, data):
+        # external location of new breadcrumb
+        if self.should_deploy(data):
+            self.locations.append(data)
```

Comment:
Why do we remember the location of external breadcrumb only when this robot would deploy a breadcrumb at said location? That does not seem right. We should remember the position every time we receive it, right?